lib/src/test/java/ECSE428Project/controller/LoginControllerTest.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.Optional;
+
+@Tag("IntegrationTest")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class LoginControllerTest {
+
+    //Variables used to create dummy account entities in the AccountRepository
+    private static final int seed = 1;
+    private static final String accountName = "accountName"+seed;
+    private static final String accountEmail = "accountEmail"+seed+"@a.ca";
+    private static final String accountPassword = "password"+seed;
+
+    private static final String wrongEmail = "wrongEmail@a.ca";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    /*
+    Create dummy test accounts in the Repo using the Account Class
+    */
+    @BeforeEach
+    public void createDummyAccount() {
+
+        //empty the repository
+        accountRepository.deleteAll();
+
+        // Create a new account with the input password, name, and email
+        Account errorAccount = new Account("wrongName", wrongEmail, "wrongPassword", false, true, 0, 0);
+        errorAccount = accountRepository.save(errorAccount);
+
+        // Create a new account with the input id, name, and email
+        Account testAccount = new Account(accountName, accountEmail, accountPassword, false, false, 0, 0);
+
+        //Save to the dummy account into the accountRepository
+        testAccount = accountRepository.save(testAccount);
+
+        //Validate the accountRepository is not empty
+        if(accountRepository.count() == 0) {
+            fail("Test Aborted: The Account Repository is empty. Entity Count = "+ accountRepository.count());
+        }
+        //Validate that the saved Account entity has: loggedIn == false
+        Assertions.assertFalse(testAccount.isLoggedIn(),
+                "Test Aborted: The dummy Account was not created successfully: LoggedIn = false");
+    }
+    /*
+   Delete dummy test accounts in the accountRepository
+   */
+    @AfterEach
+    public void clearTestingAccounts() {
+        accountRepository.deleteAll();
+        assertEquals(0, accountRepository.count());
+    }
+
+
+    @Test
+    public void testLoginSuccessful() throws Exception {
+
+        Account profile = accountRepository.findAccountByEmail(accountEmail);
+        System.out.println("At the start of the test, the Account is logged in? " + profile.isLoggedIn());
+
+        mockMvc.perform(get("/login")
+                .param("email", profile.getEmail())
+                .param("password", profile.getPassword())
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        profile = accountRepository.findAccountByEmail(accountEmail);
+        System.out.println("At the end of the test, the Account is logged in? " + profile.isLoggedIn());
```

Comment:
Maybe have a verify statement checking the account.isLoggedIn is true rather than printing out statements